easybuild/tools/run.py

```diff
@@ -77,6 +77,36 @@
 RunResult = namedtuple('RunResult', ('output', 'exit_code', 'stderr'))
 
 
+def run_cmd_cache(func):
+    """Function decorator to cache (and retrieve cached) results of running commands."""
+    cache = {}
+
+    @functools.wraps(func)
+    def cache_aware_func(cmd, *args, **kwargs):
+        """Retrieve cached result of selected commands, or run specified and collect & cache result."""
+        # cache key is combination of command and input provided via stdin
+        key = (cmd, kwargs.get('inp', None))
```

Comment:
I can't see a `inp` kwarg anywhere, where does it come from?

Reply:
Should it be `stdin`?

Reply:
Ah ok, you've renamed `inp` to `stdin` in the new `run` command. You're going to have to put in the logic to do either for now
```suggestion
        key = (cmd, kwargs.get('stdin', kwargs.get('inp', None)))
```

Reply:
Ah, that's specific to `run_cmd`, so this needs to be updated to also consider `stdin` used in `run`, nice catch!

I'll look into fixing it...

Reply:
Fixed in c045f2388, problem is caught by additional test `test_run_cache` that was added